### PR TITLE
Adds jackc/pgx instrumentation

### PR DESCRIPTION
**Pull request by jcchavezs**
## Description

This PR adds instrumentation for github.com/jackc/pgx postgres library along with some integration tests and examples.

Ping @davexroth @mohit-a21 @nxvl

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -11,6 +11,10 @@ test-unit:
 test-docker:
 	@./tests/docker/test.sh
 
+.PHONY: test-integration
+test-integration:
```
Comment (jcchavezs) [inline Makefile:15]: When we add more integration tests we can make it automatic to find the integration tests.

File: instrumentation/opentelemetry/github.com/jackc/hyperpgx/dsn.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+import (
+	"fmt"
+	"net/url"
+	"strings"
+)
+
+// parseDSN parses the connection string provided for postgres driver.
+func parseDSN(dsn string) (map[string]string, error) {
+	if !strings.HasPrefix(dsn, "postgres://") {
+		return nil, fmt.Errorf("invalid postgresql connection string: %q", dsn)
```
Comment (jcchavezs) [inline instrumentation/opentelemetry/github.com/jackc/hyperpgx/dsn.go:12]:
```suggestion
		return nil, fmt.Errorf("invalid postgres connection string as it isn't prefixed with \"postgres://\": %q", dsn)
```

File: instrumentation/opentelemetry/github.com/jackc/hyperpgx/pgx.go [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+func (w *wrappedConn) Query(ctx context.Context, query string, optionsAndArgs ...interface{}) (pgx.Rows, error) {
+	ctx, span, closer := opentelemetry.StartSpan(ctx, "db:query", &sdk.SpanOptions{Kind: sdk.Client})
+	defer closer()
+
+	for k, v := range w.connAttrs {
+		span.SetAttribute(k, v)
+	}
+	span.SetAttribute("db.statement", query)
+
+	rows, err := w.delegate.Query(ctx, query, optionsAndArgs...)
```
Comment (mohit-a21) [inline instrumentation/opentelemetry/github.com/jackc/hyperpgx/pgx.go:69]: is the response of no use to us?

**Review by mohit-a21 — APPROVED** [review]

File: instrumentation/opentelemetry/github.com/jackc/hyperpgx/pgx.go (hunk 3 above)
Comment (davexroth) [reply instrumentation/opentelemetry/github.com/jackc/hyperpgx/pgx.go:69]: for sql, we don't need to capture the response.  we're only interested in the query.

File: instrumentation/opentelemetry/github.com/jackc/hyperpgx/dsn.go [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+	}
+
+	parsedURL, err := url.Parse("http://" + dsn[11:])  // 11 = len("postgres://")
```
Comment (jcchavezs) [inline instrumentation/opentelemetry/github.com/jackc/hyperpgx/dsn.go:15]:
```suggestion
	// we prepend "http://" to be able to parse the connection string as a URL as it has the same structure.
	parsedURL, err := url.Parse("http://" + dsn[11:])  // 11 = len("postgres://")
```

[the remark this answers is not in the document]
Comment (jcchavezs) [reply instrumentation/opentelemetry/github.com/jackc/hyperpgx/dsn.go:16]: We prepend the http just to get the standard library to parse it as a URL as the structure is the same.

[the remark this answers is not in the document]
Comment (jcchavezs) [reply instrumentation/opentelemetry/github.com/jackc/hyperpgx/dsn.go]: nvm I removed this.

**Pull request merged**